<?xml version="1.0" encoding="UTF-8"?>
<interface>
  <requires lib="gtk" version="4.0"/>
  <template class="GsReviewRow" parent="GtkListBoxRow">
    <property name="activatable">False</property>
    <style>
      <class name="review-row"/>
    </style>
    <child>
      <object class="GtkBox">
        <property name="orientation">vertical</property>
        <property name="spacing">6</property>
        <property name="hexpand">True</property>
        <property name="vexpand">True</property>
        <child>
          <object class="GtkBox">
            <property name="spacing">10</property>
            <property name="hexpand">True</property>
            <child>
              <object class="GsStarWidget" id="stars">
                <property name="halign">start</property>
                <property name="sensitive">False</property>
              </object>
            </child>
            <child>
              <object class="GtkLabel" id="summary_label">
                <property name="halign">start</property>
                <property name="label">Steep learning curve, but worth it</property>
                <property name="ellipsize">end</property>
                <property name="hexpand">True</property>
                <property name="selectable">True</property>
                <style>
                  <class name="heading"/>
                </style>
              </object>
            </child>
            <child>
              <object class="GtkLabel" id="date_label">
                <property name="halign">end</property>
                <property name="label">3 January 2016</property>
                <property name="hexpand">True</property>
                <property name="selectable">True</property>
                <style>
                  <class name="dim-label"/>
                </style>
              </object>
            </child>
          </object>
        </child>
        <child>
          <object class="GtkLabel" id="author_label">
            <property name="halign">start</property>
            <property name="label">Angela Avery</property>
            <property name="ellipsize">end</property>
            <property name="selectable">True</property>
            <style>
              <class name="dim-label"/>
            </style>
          </object>
        </child>
        <child>
          <object class="GtkLabel" id="text_label">
            <property name="halign">start</property>
            <property name="margin_top">10</property>
            <property name="margin_bottom">8</property>
            <property name="label">Best overall 3D app I've ever used overall 3D app I've ever used. Best overall 3D app I've ever used overall 3D app I've ever used. Best overall 3D app I've ever used overall 3D app I've ever used. Best overall 3D app I've ever used overall 3D app I've ever used.</property>
            <property name="wrap">True</property>
            <property name="max_width_chars">80</property>
            <property name="xalign">0</property>
            <property name="wrap-mode">word-char</property>
            <property name="selectable">True</property>
          </object>
        </child>
        <child>
          <object class="GtkBox">
            <property name="visible">True</property>
            <property name="spacing">9</property>
            <child>
              <object class="GtkLabel" id="your_review_label">
                <property name="valign">center</property>
                <property name="label" translatable="yes">Your review</property>
                <property name="halign">start</property>
                <property name="xalign">0</property>
                <property name="yalign">0</property>
		<property name="visible">False</property>
                <style>
                  <class name="installed-badge"/>
                </style>
              </object>
            </child>
            <child>
              <object class="GtkBox" id="box_voting">
                <property name="visible">False</property>
                <property name="spacing">9</property>
                <child>
                  <object class="GtkBox" id="box_vote_buttons">
                    <style>
                      <class name="vote-buttons"/>
                    </style>
                    <child>
                      <object class="GtkButton" id="button_like">
                        <property name="tooltip-text" translatable="yes">Like Review</property>
                        <property name="icon-name">review-rate-positive-symbolic</property>
                        <property name="can_focus">True</property>
                        <property name="receives_default">True</property>
                        <property name="halign">end</property>
                        <style>
                          <class name="flat"/>
                        </style>
                      </object>
                    </child>
                    <child>
                      <object class="GtkButton" id="button_dislike">
                        <property name="tooltip-text" translatable="yes">Dislike Review</property>
                        <property name="icon-name">review-rate-negative-symbolic</property>
                        <property name="can_focus">True</property>
                        <property name="receives_default">True</property>
                        <property name="halign">end</property>
                        <style>
                          <class name="flat"/>
                        </style>
                      </object>
                    </child>
                  </object>
                </child>
              </object>
            </child>
            <child>
              <object class="GtkButton" id="button_report">
                <property name="tooltip-text" translatable="yes">Report…</property>
                <property name="icon-name">review-report-symbolic</property>
                <property name="can_focus">True</property>
                <property name="receives_default">True</property>
                <property name="halign">end</property>
                <property name="hexpand">True</property>
                <style>
                  <class name="flat"/>
                </style>
              </object>
            </child>
            <child>
              <object class="GtkButton" id="button_remove">
                <property name="tooltip-text" translatable="yes">Remove…</property>
                <property name="icon-name">user-trash-symbolic</property>
                <property name="can_focus">True</property>
                <property name="receives_default">True</property>
                <property name="halign">end</property>
                <property name="hexpand">True</property>
                <style>
                  <class name="flat"/>
                  <class name="destructive-action"/>
                </style>
              </object>
            </child>
          </object>
        </child>
      </object>
    </child>

  </template>

  <object class="GtkSizeGroup" id="action_sizegroup">
    <widgets>
      <widget name="button_report"/>
      <widget name="button_remove"/>
    </widgets>
  </object>
  <object class="GtkSizeGroup" id="useful_sizegroup">
    <widgets>
      <widget name="button_like"/>
      <widget name="button_dislike"/>
    </widgets>
  </object>

</interface>
